Tools/travis_compile.sh

```diff
@@ -3,6 +3,27 @@ set -xe
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
+mkdir -p sdkcache/
+git submodule | cut -d'(' -f1 | sort > sdkcache/current.txt
```

Comment:
git submodule status  --recursive so you can get all the submodule in  submodules